pageserver/src/tenant.rs

```diff
@@ -1393,6 +1393,32 @@ impl Tenant {
         Ok(tl)
     }
 
+    /// Helper for unit tests to create a timeline with some pre-loaded states.
+    #[cfg(test)]
+    #[allow(clippy::too_many_arguments)]
+    pub async fn create_test_timeline_with_states(
```

Comment:
Let's call this `with_layers` not states, because we already have (too many) states with TimelineState, TenantState, TenantManagerState.